projects/fsastorefrontlib/src/core/my-account/store/effects/claim.effect.ts

```diff
@@ -156,6 +156,38 @@ export class ClaimEffects {
     })
   );
 
+  @Effect()
+  changeClaim$: Observable<any> = this.actions$.pipe(
+    ofType(fromActions.CHANGE_CLAIM),
+    map((action: fromActions.ChangeClaim) => action.payload),
+    switchMap(payload => {
+      const claimNumber = payload.claimData.claimNumber;
+      return this.claimConnector
+        .updateClaim(payload.userId, claimNumber, payload.claimData)
+        .pipe(
+          map(claim => {
+            this.routingService.go({
+              cxRoute: 'claimDetails',
+              params: { claimId: claimNumber },
+            });
+            this.globalMessageService.add(
+              {
+                key: 'claim.successfulChangeClaim',
+                params: {
+                  claimNumber: claimNumber,
+                },
+              },
+              GlobalMessageType.MSG_TYPE_CONFIRMATION
+            );
+            return new fromActions.UpdateClaimSuccess(claim);
+          }),
+          catchError(error =>
+            of(new fromActions.UpdateClaimFail(JSON.stringify(error)))
+          )
+        );
+    })
+  );
+
```

Comment:
Should we have logic in reducer that handles this new action?

Reply:
No need, updateclaimsuccess is doing what we need. The result of changing claim should be a succesifull update. I would keep it as is.